### PR TITLE
[#592] Describe configuration

### LINKED ISSUE #592: Describe configuration in doc file
See [neo-go/docs/node-configuration.md](https://github.com/nspcc-dev/neo-go/blob/master/docs/node-configuration.md)

Ref: https://github.com/nspcc-dev/neofs-node/pull/1604

### PR DESCRIPTION
**Pull request by KirillovDenis**
close #592 
Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -34,18 +34,19 @@ a gateway spread requests equally among them (using weight 1 for every node):
...
 
-Wallet (`--wallet`) is a mandatory parameter. It is a path to a wallet file. You can provide a passphrase to decrypt 
+Wallet (`--wallet`) is a mandatory parameter. It is a path to a wallet file. You can provide a passphrase to decrypt
 a wallet via env variable or conf file, or you will be asked to enter a password interactively.
 You can also specify an account address to use from a wallet using the `--address` parameter.
 
 ### Listening on address and TLS
 
-Gateway listens on `0.0.0.0:8080` by default, and you can change that with the `--listen_address` option.
+You can make the gateway listen on specific address using the `--listen_address` option.
```
Comment (masterSplinter01) [inline docs/configuration.md:49]: I agree that this wording sounds better, but why you decided to remove a mention to a default value?

File: docs/configuration.md [hunk 2]
```diff
@@ -100,30 +100,227 @@ default. To enable them, use `--pprof` and `--metrics` flags or
```
Comment (masterSplinter01) [inline docs/configuration.md:100]: To be more consistent, shouldn't we remove mention to a default value in Monitoring and metrics?
` Pprof and Prometheus are integrated into the gateway, but not enabled by
default.`

**Review by alexvanin — COMMENTED** [review]
Back in the day @532910 noticed that `config.yaml` file defines
```yaml
resolve_order:
  - nns
```
but default resolve order is actual `-dns` only. What do you think about fixing it?

**Comment by KirillovDenis** [comment]
Currently, not all params in config file represent defaults. So I see no criminal here. But we can fix this particular config value.

File: docs/configuration.md [hunk 3]
```diff
@@ -72,58 +73,253 @@ $ neofs-s3-gw --rpc_endpoint http://morph-chain.neofs.devenv:30333/ --resolve_or
...
+| `address`              | `string`   |                | Account address to get from wallet. If omitted default one will be used.                                                                                                                                          |
+| `listen_address`       | `string`   | `0.0.0.0:8080` | The address that the gateway is listening on.                                                                                                                                                                     |
+| `rpc_endpoint`         | `string`   |                | The address of the RPC host to which the gateway connects to resolve bucket names.                                                                                                                                |
+| `resolve_order`        | `[]string` | `[dns]`        | Order of bucket name resolvers to use.                                                                                                                                                                            |
```
Comment (alexvanin) [inline docs/configuration.md:164]: > Available resolvers: dns, nns.

**Review by alexvanin — APPROVED** [review]

**Comment by alexvanin** [comment]
Can you mention tree service parameter in configs until it's gone after #609 ?

File: docs/configuration.md [hunk 4]
```diff
@@ -72,58 +73,265 @@ $ neofs-s3-gw --rpc_endpoint http://morph-chain.neofs.devenv:30333/ --resolve_or
...
+
+| Parameter | Type     | Default value | Description                                                                                        |
+|-----------|----------|---------------|----------------------------------------------------------------------------------------------------|
+| `level`   | `string` | `debug`       | Logging level.<br/>Possible values:  `debug`, `info`, `warn`, `error`, `dpanic`, `panic`, `fatal`. |
+
+### `tree` section
+
+```yaml
+tree:
+  service: s01.neofs.devenv:8080
```
Comment (masterSplinter01) [inline docs/configuration.md:245]: Let's add this example to `config.env` and `config.yaml` too

**Review by alexvanin — APPROVED** [review]

**Pull request merged**